capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java

```diff
@@ -1,166 +1,134 @@
+          System.err.println("Error: no such document!");
+        }
+      } catch (Exception e) {
+        System.err.println("Error: " + e);
+      }
+      result.add(club);
+    }
+    else {
+      ApiFuture<QuerySnapshot> future = clubs.get();
+      try {
+        List<QueryDocumentSnapshot> documents = future.get().getDocuments();
+        for (QueryDocumentSnapshot document : documents) {
+          result.add(document.toObject(Club.class));
+        }
+      } catch (Exception e) {
+        System.err.println("Error: " + e);
+      }
     }
-    Gson gson = new Gson();
     response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(clubs));
+    response.getWriter().println(gson.toJson(result));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    JsonObject jsonObject = createRequestBodyJson(request);
+    JsonObject jsonObject = Utility.createRequestBodyJson(request);
+    String id = jsonObject.get("id").getAsString();
     String name = jsonObject.get("name").getAsString();
-    String description = jsonObject.get("description").getAsString();
-    long ownerID = jsonObject.get("ownerID").getAsLong();
-
-    String announcement = "";
-    if (jsonObject.get("announcement") != null) {
-      announcement = jsonObject.get("announcement").getAsString();
-    }
+    String ownerID = jsonObject.get("ownerID").getAsString();
+    String description = "";
+    String gbookID = "";
 
-    Collection<Long> inviteIDs = new ArrayList<>();
-    if (jsonObject.get("inviteIDs") != null) {
-      for(JsonElement j : jsonObject.get("inviteIDs").getAsJsonArray()) {
-        inviteIDs.add(Long.parseLong(j.getAsString()));
-      } 
+    if (jsonObject.get("description") != null) {
+      description = jsonObject.get("description").getAsString();
     }
-
-    String gbookID = "";
     if (jsonObject.get("gbookID") != null) {
       gbookID = jsonObject.get("gbookID").getAsString();
     }
 
-    Entity clubEntity = new Entity("Club");
-
-    clubEntity.setProperty("id", clubEntity.getKey().getId());
-    clubEntity.setProperty("name", name);
-    clubEntity.setProperty("description", description);
-    clubEntity.setProperty("announcement", announcement);
-    clubEntity.setProperty("posts" , new ArrayList<String>());
-    clubEntity.setProperty("ownerID", ownerID);
-    clubEntity.setProperty("inviteIDs", inviteIDs);
-    clubEntity.setProperty("memberIDs", new ArrayList<Long>());
-    clubEntity.setProperty("gbookID", gbookID);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(clubEntity);
+    Club club = new Club(id, name, description, ownerID, gbookID);
+    ApiFuture<WriteResult> future = clubs.document(id).set(club);
+    try {
+      System.out.println("Update time : " + future.get().getUpdateTime());
+    } catch (Exception e) {
+      System.err.println("Error: " + e);
```

Comment:
Add some logic for the case of I/O failure here. You could optionally retry the request and ultimately respond with an error code if needed.

Reply:
That's a good idea. I'm not sure how to trigger a "retry" but I can look into it for future iterations, for now I'll just send a 500. I just want to get the base functionality down, as I'll be abstracting much of this in the next PR.